add did:nda method

thanhhuykhtn opened the pull request:
This PR registers the did:nda method in the W3C DID Method Registry.

Specification: https://did-specs.ndachain.vn/
Contact: Nguyen Thanh Huy ([contact@ndachain.vn](mailto:contact@ndachain.vn))
Verifiable Data Registry: NDA Chain Registry

### DID Method Registration

As a DID method registrant, I have ensured that my DID method registration complies with the following statements:

- [x] The DID Method specification [defines the DID Method Syntax](https://w3c.github.io/did-core/#method-syntax).
- [x] The DID Method specification [defines the Create, Read, Update, and Deactivate DID Method Operations](https://w3c.github.io/did-core/#method-operations).
- [x] The DID Method specification [contains a Security Considerations section](https://w3c.github.io/did-core/#security-requirements).
- [x] The DID Method specification [contains a Privacy Considerations section](https://w3c.github.io/did-core/#privacy-requirements).
- [x] The JSON file I am submitting has [passed all automated validation tests below](#partial-pull-merging).
- [x] The JSON file contains a `contactEmail` address [OPTIONAL].
- [x] The JSON file contains a `verifiableDataRegistry` entry [OPTIONAL].

swcurran commented:
Hi @thanhhuykhtn -- Thanks for the updates.  The clarifications help -- thank you.  I do have a question from the perspective of an author of a DID Method Spec.  When writing our DID Method, a crucial goal was to allow others to write their own implementation of a DID Registrar / Controller and a DID Resolver.  So we tried to be very clear on what an implementer would have to do to create their own implementation.  If you have the same goal, you might consider adding to the spec from that perspective.  For example, in the `create` section, you note that given a key pair, you know the DID identifier. How is that done?  I'm guessing that the use of the NDAChain creates limitations on the cryptographic key types that can be used. What are they?  That can be stated implicitly, or pointed to in another document.  A pointer is good if that data is likely to change in the future.

That said, I think the spec is sufficient for registration.  These comments are more about making the spec more useful for anyone wanting to interact with a `did:nda`.

thanhhuykhtn commented:
Dear @swcurran @ottomorac  

Thanks again for the great feedback!

I have updated the [DID Method Spec ](https://did-specs.ndachain.vn/)as your suggestions:
- Added detailed guidance on key pair generation algorithms and supported key types
- Expanded the Create section with step-by-step instructions for:
  + Generating compliant key pairs and deriving the address
  + Creating the DID from the public key
  + Registering the DID with NDAChain's 
- Included implementation notes for building independent registrars/resolvers

This should now make it much clearer for anyone wanting to implement did:nda from scratch.

Let me know if anything else is needed for merge!
